File: python/websocket/producer.py
import websocket
import base64
import json

pulsarURL = '{{ pulsar_url }}'
pulsarJWT = '{{ pulsar_jwt }}'
pulsarTopic = 'persistent/public/default/topic-go-pandio-go/'

ws = websocket.create_connection(pulsarURL + pulsarTopic, header=["Authorization: Bearer "+pulsarJWT])
# Send one message as JSON
ws.send(json.dumps({
    'payload': base64.b64encode(b'Hello world!').decode('utf-8'),
}))

response = json.loads(ws.recv())
if response['result'] == 'ok':
    print("received ack: {}".format(response))
else:
    print('Failed to publish message:', response)
ws.close()

File: nodejs/websocket/producer.js
const path = require('path');
const WebSocket = require('ws');

const pulsarURL = "{{ pulsar_url }}"
const pulsarJWT = "{{ pulsar_jwt }}"
const pulsarTopic = "persistent/{{pulsar_tenant}}/{{pulsar_namespace}}/nodejs-websocket"

const topic = path.join(pulsarURL, '/v2/producer/', pulsarTopic);
const ws = new WebSocket(topic, { headers: { Authorization: "Bearer  " + pulsarJWT } });

var message = {
	"payload": new Buffer.from("Hello World!").toString('base64')
};

ws.on('open', function () {
	// Send one message
	ws.send(JSON.stringify(message));
});

ws.on('message', function (message) {
	console.log('received ack: %s', message);
	ws.close()
});

ws.on("error", function (message) {
	console.log('received error: %s', message);
});
